hivemind/client/optim/simple.py

```diff
@@ -0,0 +1,88 @@
+from threading import Thread, Lock, Event
+from typing import Optional, Callable, Any
+
+import torch
+
+from hivemind.dht import DHT
+from hivemind.client.averaging import DecentralizedAverager
+from hivemind.client.optim.base import DecentralizedOptimizerBase
+from hivemind.utils import MPFuture, get_logger
+
+logger = get_logger(__name__)
+
+
+class DecentralizedOptimizer(DecentralizedOptimizerBase):
+    """
+    A simple optimizer that trains a shared model by averaging model parameters with peers in the background.
+
+    :param opt: a pytorch optimizer configured to update model parameters.
+    :param dht: a running hivemind DHT daemon connected to other peers
+    :param prefix: all DHT keys that point to optimization metadata will have this prefix
+    :param target_group_size: maximum group size for averaging (see DecentralizedAverager)
+    :param kwargs: additional parameters passed to DecentralizedAverager
```

Comment:
We need an explicit argument for update frequency in the __init__ for this class, so that it isn't necessary to look for the frequency in other docstrings

Reply:
Solution: implemented averaging_period, measured in seconds. Here's how it works:

If the optimizer is active, the background thread will await until get_dht_time is divisible by averaging_period, then begin averaging.

This trick ensures that all peers begin averaging at almost the same time. Otherwise, having a long averaging_period will result in a situation where only a small fraction of peers is active at any given point in time and no peer is able to find group (to say nothing about following the moshpit protocol)